### PR TITLE
Counting pending tuples

### PR DESCRIPTION
**Pull request by kalimfaria**
Added a metric that counts the number of inbound packets received at the stream manager that need to be sent to their destination instances @huijunw @nwangtw @nlu90

### CODE REVIEW

File: heron/stmgr/src/cpp/manager/instance-server.cpp [hunk 1]
```diff
@@ -154,8 +157,23 @@ InstanceServer::~InstanceServer() {
     }
   }
 
+  for (auto qlmIter = connection_buffer_length_metric_map_.begin();
+      qlmIter != connection_buffer_length_metric_map_.end(); ++qlmIter) {
+    const sp_string& instance_id = qlmIter->first;
+    for (auto iter = instance_info_.begin(); iter != instance_info_.end(); ++iter) {
```
Comment (nwangtw) [inline heron/stmgr/src/cpp/manager/instance-server.cpp]: suggest to refactor this code into a findInstance() or findFirstInstance() function.

File: heron/stmgr/src/cpp/manager/instance-server.h [hunk 2]
```diff
@@ -163,6 +164,14 @@ class InstanceServer : public Server {
   typedef std::unordered_map<sp_string, heron::common::MultiMeanMetric*> ConnectionBufferMetricMap;
   ConnectionBufferMetricMap connection_buffer_metric_map_;
 
+  // map of Instance_id to queue length metric
+  typedef std::unordered_map<sp_string, heron::common::MultiCountMetric*>
+   ConnectionBufferLengthMetricMap;
```
Comment (nwangtw) [inline heron/stmgr/src/cpp/manager/instance-server.h]: bad indent

File: heron/stmgr/src/cpp/manager/instance-server.cpp [hunk 3]
```diff
@@ -154,8 +157,23 @@ InstanceServer::~InstanceServer() {
     }
   }
 
+  for (auto qlmIter = connection_buffer_length_metric_map_.begin();
```
Comment (nwangtw) [inline heron/stmgr/src/cpp/manager/instance-server.cpp]: a little complicated, please add comment.

Comment (kalimfaria) [reply heron/stmgr/src/cpp/manager/instance-server.cpp]: comment added and simplified in next commit.

File: heron/stmgr/src/cpp/manager/instance-server.cpp (hunk 1 above)
Comment (kalimfaria) [reply heron/stmgr/src/cpp/manager/instance-server.cpp]: simplified code in next commit

File: heron/stmgr/src/cpp/manager/instance-server.h (hunk 2 above)
Comment (kalimfaria) [reply heron/stmgr/src/cpp/manager/instance-server.h]: fixed in next commit.

**Comment by huijunw** [comment]
ship after ci pass

**Comment by nwangtw** [comment]
Hmm. Unit test failure. @kalimfaria Please check.

**Comment by kramasamy** [comment]
@kalimfaria - can you please take care of the failing test? Once the ci passes we can merge it.

**Pull request merged**